src/App.tsx

```diff
@@ -1,61 +1,93 @@
+import { useState } from 'react';
+
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { NewTodo, Todo } from './types/Todo';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const preTodos = todosFromServer.map((todo: Todo) => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId) as User,
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<NewTodo[]>(preTodos);
+  const [title, setTitle] = useState('');
+  const [hasTitle, setHasTitle] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserId, setHasUserId] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setHasTitle(!title);
+    setHasUserId(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo = {
+      id: Math.max(...preTodos.map(todo => todo.id)) + 1,
+      title: title,
+      completed: false,
+      userId: userId,
+      user: usersFromServer.find(user => user.id === userId) as User,
+    };
+
+    setTodos(prevState => [...prevState, newTodo]);
+
+    setTitle('');
+    setUserId(0);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={event => {
+              setTitle(event.target.value);
+              setHasTitle(false);
+            }}
+          />
+          {hasTitle && <span className="error">Please enter a title</span>}
```

Comment:
Sounds strange a bit 'if the title field has title, please enter a title'. Try to rename this variable somehow, `isTitleEmpty` for instance or stuff like that